web/src/lib/components/asset-viewer/asset-viewer.svelte

```diff
@@ -250,68 +250,22 @@
     isShowActivity = !isShowActivity;
   };
 
-  const handleKeypress = async (event: KeyboardEvent) => {
-    if (shouldIgnoreShortcut(event)) {
-      return;
-    }
-
-    const key = event.key;
-    const shiftKey = event.shiftKey;
-    const ctrlKey = event.ctrlKey;
-
-    if (ctrlKey) {
-      return;
-    }
-
-    switch (key) {
-      case 'a':
-      case 'A': {
-        if (shiftKey) {
-          await toggleArchive();
-        }
-        return;
-      }
-      case 'ArrowLeft': {
-        await navigateAsset('previous');
-        return;
-      }
-      case 'ArrowRight': {
-        await navigateAsset('next');
-        return;
-      }
-      case 'd':
-      case 'D': {
-        if (shiftKey) {
-          await downloadFile(asset);
-        }
-        return;
-      }
-      case 'Delete': {
-        await trashOrDelete(shiftKey);
-        return;
-      }
-      case 'Escape': {
-        if (isShowDeleteConfirmation) {
-          isShowDeleteConfirmation = false;
-          return;
-        }
-        if (isShowShareModal) {
-          isShowShareModal = false;
-          return;
-        }
-        closeViewer();
-        return;
-      }
-      case 'f': {
-        await toggleFavorite();
-        return;
-      }
-      case 'i': {
-        isShowActivity = false;
-        $isShowDetail = !$isShowDetail;
-        return;
-      }
-    }
+  let shortcuts: ShortcutList;
+  $: shortcuts = [
+    [{ key: 'a', shift: true }, () => toggleArchive()],
+    [{ key: 'ArrowLeft' }, () => navigateAsset('previous')],
+    [{ key: 'ArrowRight' }, () => navigateAsset('next')],
+    [{ key: 'd', shift: true }, () => downloadFile(asset)],
```

Comment:
Can some shortcuts be case insensitive like before ?

Reply:
Matching shortcuts is case insensitive, this is handled with `shortcut.key.toLowerCase() === event.key.toLowerCase()`

Reply:
Oh nice !